packages/cli/src/commands/clean.ts

```diff
@@ -17,143 +17,126 @@ import { logger } from '@salto-io/logging'
 import { EOL } from 'os'
 import { loadLocalWorkspace, cleanWorkspace } from '@salto-io/core'
 import { WorkspaceComponents } from '@salto-io/workspace'
-import { ParsedCliInput, CliOutput, CliExitCode, CliCommand, CliTelemetry } from '../types'
-import { createCommandBuilder } from '../command_builder'
-import { getCliTelemetry } from '../telemetry'
 import { getWorkspaceTelemetryTags } from '../workspace/workspace'
 import { errorOutputLine, outputLine } from '../outputer'
 import { getUserBooleanInput } from '../callbacks'
 import { formatCleanWorkspace, formatCancelCommand, header, formatStepStart, formatStepFailed, formatStepCompleted } from '../formatter'
 import Prompts from '../prompts'
+import { CliExitCode } from '../types'
+import { createPublicCommandDef, DefActionInput } from '../command_builder'
 
 const log = logger(module)
 
-type CleanArgs = WorkspaceComponents & { force: boolean }
+type CleanArgs = {
+    force: boolean
+} & WorkspaceComponents
 
-type CleanParsedCliInput = ParsedCliInput<CleanArgs>
+const action = async (
+  {
+    input: { force, ...cleanArgs },
+    cliTelemetry,
+    output,
+    workingDir = '.',
+  }: DefActionInput<CleanArgs>
+): Promise<CliExitCode> => {
```

Comment:
nit: instead of exporting `DefActionInput`, export `CommandDefAction` and then this definition could be a little cleaner
```suggestion
const action: CommandDefAction<CleanArgs> = async ({
  input: { force, ...cleanArgs },
  cliTelemetry,
  output,
  workingDir = '.',
}) => {
```